src/sentry/integrations/discord/actions/metric_alert.py

```diff
@@ -63,13 +64,15 @@ def send_incident_alert_notification(
         try:
             client.send_message(channel, message)
         except Exception as error:
-            # TODO(iamrajjoshi): Update some of these failures to halts
             lifecycle.add_extras(
                 {
                     "incident_id": incident.id,
                     "channel_id": channel,
                 }
             )
-            lifecycle.record_failure(error)
+            if isinstance(error, ApiRateLimitedError):
```

Comment:
While I largely agree that this is a user-caused issue now, this might be problematic for users with larger orgs in the future if they decide to use the discord integration. We don't have to address this now, but we may want to consider having some mechanism to backlog and batch in the future (if possible).